src/app/order/OrderStatus.tsx

```diff
@@ -11,27 +11,18 @@ export interface OrderStatusProps {
     order: Order;
 }
 
+type PaymentWithMandate = GatewayOrderPayment & Required<Pick<GatewayOrderPayment, 'mandate' | 'methodId'>>;
+
+const isPaymentWithMandate = (payment: GatewayOrderPayment | GiftCertificateOrderPayment): payment is PaymentWithMandate =>
+    !!payment.methodId && ('mandate' in payment && !!payment.mandate);
```

Comment:
once you have the updated type from the sdk, this can just be:

`!!payment.methodId && !!payment.mandate`

Reply:
@leeBigCommerce I tired to make a build with the updated sdk and it's throwing the error
```
TS2339: Property ‘mandate’ does not exist on type ‘GiftCertificateOrderPayment | GatewayOrderPayment’.
      Property ‘mandate’ does not exist on type ‘GiftCertificateOrderPayment’. 
```

I will revert to the  `'mandate' in payment`.

Reply:
Makes sense, the current type definitions don't help us out in sorting `GiftCertificateOrderPayment`s from `GatewayOrderPayment`s